### PR TITLE
Feature/start route

### PR DESCRIPTION
**Pull request by dephiloper**
Added explanatory comments to the segment navigator as introduction it should be used.
When there are open questions just ask here or in slack.

### CODE REVIEW

File: lib/pages/setup/route_preview.dart [hunk 1]
```diff
@@ -171,3 +172,5 @@ class _RoutePreviewState extends State<RoutePreview> {
     if (_routes.length > 0) switchRoute(_currentRouteIndex);
```
Comment (jakubprogramming) [inline lib/pages/setup/route_preview.dart]: We is this method (onMapReady) still in here? I cannot find any place where it is called.
Should probably be deleted before merge?!

File: lib/components/map_widget.dart [hunk 2]
```diff
@@ -12,9 +12,8 @@ import 'package:location_permissions/location_permissions.dart';
 
 class MapWidget extends StatefulWidget {
   final bool isStatic;
-  final VoidCallback onMapReady;
 
-  MapWidget({Key key, @required this.isStatic, this.onMapReady})
```
Comment (jakubprogramming) [inline lib/components/map_widget.dart:17]: seems linke onMapReady is not needed anymore?!

File: lib/components/map_widget.dart [hunk 3]
```diff
@@ -138,8 +137,6 @@ class MapWidgetState extends State<MapWidget> {
       _lineRoute = lineRoute;
       _linePassedRoute = linePassedRoute;
     });
-
-    if (!widget.isStatic) initUpdateRouteTimer();
```
Comment (jakubprogramming) [inline lib/components/map_widget.dart]: Ok so, you have deleted the initUpdateRouteTimer call, but you have not added a call at any other place.
Thus our routes will not be updating anymore, making "navigation" impossible

Comment (jakubprogramming) [reply lib/components/map_widget.dart]: Actually tried it out, routes no longer update (how could they?) :D

File: lib/navigation/segment_navigator.dart [hunk 4]
```diff
@@ -17,47 +18,50 @@ class SegmentRoutes {
   static const String credits = '/more/credits';
 }
 
+/// There are two options for navigation between pages.
```
Comment (jakubprogramming) [inline lib/navigation/segment_navigator.dart:21]: those comments will definitely be helpful!

File: lib/navigation/segment_navigator.dart [hunk 5]
```diff
@@ -66,19 +70,65 @@ class SegmentNavigator extends StatelessWidget {
...
+  Map<String, WidgetBuilder> _routeBuilders(BuildContext context,
+      [Map<String, dynamic> params]) {
+    return {
+      SegmentRoutes.credits: (context) => CreditsPage(),
+
+      SegmentRoutes.help: (context) => HelpPage(
+          onPushHistorySaveState: () => onChangeSegment(AppSegment.history),
+          onPushHistory: () => onChangeSegment(AppSegment.history, true)),
+
+      // TODO add route preferences
```
Comment (jakubprogramming) [inline lib/navigation/segment_navigator.dart:83]: what's up with these 2 TODOs? 
Are they here until we have the corresponding pages?

**Review by jakubprogramming — CHANGES_REQUESTED** [review]
Overall I like your changes. I do not entirely understand the navigation flow yet, but I am confident that the next time I need to work with it your comments will help me to get around. 

Mostly great stuff. You did break the functionality of updating routes though, and there is a method that is not needed anymore!? Just read my comments. 

Will approve after changes have been discussed and error (routes not updating) is fixed.

File: lib/components/map_widget.dart (hunk 2 above)
Comment (dephiloper) [reply lib/components/map_widget.dart:17]: done

File: lib/navigation/segment_navigator.dart (hunk 5 above)
Comment (dephiloper) [reply lib/navigation/segment_navigator.dart:83]: correct, they are for adjusting the other pages later.

File: lib/pages/setup/route_preview.dart (hunk 1 above)
Comment (dephiloper) [reply lib/pages/setup/route_preview.dart]: done

File: lib/components/map_widget.dart (hunk 3 above)
Comment (dephiloper) [reply lib/components/map_widget.dart]: you are right, dunno why this was deleted..

**Comment by jakubprogramming** [comment]
alright. You removed the unnecessary method and re-added the call to initialise the updateRouteTimer. For some reasons in my testing routes are not correctly updated anymore, even though everything is called correctly. 

I will investigate the issue, but I will do this in a different branch where I will furthermore improve the updateRoute mechanism.

**Review by jakubprogramming — APPROVED** [review]

**Pull request merged**